docs/handling-eviction-requests.md

```diff
@@ -0,0 +1,79 @@
+```shell
+kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io virt-api-validator -o yaml
+```
+
+It has the ability to admit eviction requests before `kube-api` checks them against [Pod Distribution Budget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets) objects.
+
+In case the pod is not a `virt-launcher` pod - the eviction request is approved.
+Otherwise, the webhook takes action according to the VMI's eviction strategy:
+
+| Eviction Strategy     | Is VMI migratable | Is VMI marked for eviction | Does Webhook approve eviction | Response               |
+|-----------------------|-------------------|----------------------------|-------------------------------|------------------------|
+| None                  | N/A               | False                      | True                          | 200 - Eviction granted |
+| LiveMigrate           | True              | True                       | True                          | 200 - Eviction granted |
+| LiveMigrate           | False             | False                      | False                         | 429 - Eviction denied  |
+| LiveMigrateIfPossible | True              | True                       | True                          | 200 - Eviction granted |
+| LiveMigrateIfPossible | False             | False                      | True                          | 200 - Eviction granted |
+| External              | N/A               | True                       | True                          | 200 - Eviction granted |
+
+The webhook sets the value of the VMI's `Status.EvacuationNodeName` to the node name its currently running on, so the evacuation controller will know it needs to migrate it to another node.
+
+# Pod Distribution Budget
+In order for the eviction of VMIs to happen in a controlled manner, KubeVirt protects part of the `virt-launcher` pods with a PDB which blocks eviction requests.
+
+`virt-controller` has a `Disruption Budget Controller` which decides whether a `virt-launcher` pod should be protected based on the eviction strategy of its controlling VMI:
+
+| Eviction Strategy     | Is a PDB required             | 
+|-----------------------|-------------------------------|
+| None                  | False                         |
+| LiveMigrate           | True                          |
+| LiveMigrateIfPossible | Only if the VMI is migratable |
+| External              | True                          |
+
+# Eviction Approval Summery
+
+The eviction request's initiator will observe one of the following responses:
+
+| Eviction Strategy     | Is VMI migratable | Is VMI marked for eviction | Does Webhook approve eviction | Does PDB allow eviction | Response                         |
+|-----------------------|-------------------|----------------------------|-------------------------------|-------------------------|----------------------------------|
```

Comment:
How should be the table read? Either as
- If "Eviction strategy None" then ("Is VMI migratable" does not apply, "VM is not marked for eviction", "Webhook approves eviction", ...)

or

- If ("Eviction strategy" == None, "Is VMI migratable" can be of any value, "Is VMI marked for eviction" == false, "Does Webhook approve eviction" == true, "Does PDB allow eviction" == true) then Response

or differently?

Reply:
Depending on the eviction strategy and is VMI migratable - the webhook decides what to do.
Both the webhook and the PDB should approve the eviction in order to receive 200, otherwise 429 is returned.